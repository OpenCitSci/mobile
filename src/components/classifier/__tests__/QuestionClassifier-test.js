import React from 'react'
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux'
import {createStore} from 'redux'

const store = createStore(() => {
}, ['test'])

import { mapStateToProps, QuestionClassifier} from '../QuestionClassifier'
import {AnswerButton} from '../ClassifierButton'
import {getAnswersFromWorkflow, getTaskFromWorkflow} from '../../../utils/workflow-utils';


const subject = {
    id: '23432432',
    display: {
        src: 'blah.jpg'
    }
}

const project = {
    display_name: 'Awesome project'
}

const subjectSizes = {resizedWidth: 100, resizedHeight: 100}


const seenThisSession = []

// Stub out actions
const navBarActions = {
    setTitleForPage() {
    },
    setColorForPageToDefault() {
    }
}

const classifierActions = {
    startNewClassification() {
    },
    setClassifierTestMode() {
    }
}

const task = {
    question: 'Do you like this question?',
    answers: [
        {label: 'Yes'},
        {label: 'No'},
        {label: 'Maybe - I have to think about it'},
    ]
}

const workflow = {
    first_task: 'T0',
    tasks: {
        T0: task
    },
    configuration: {
        pan_and_zoom: true
    }
}

// class TestQuestionClassifier extends QuestionClassifier {
//
const initialState = {
    task: task,
    answers: getAnswersFromWorkflow(workflow),
    classifier: {
        isSuccess: true,
        isFailure: false,
        isFetching: false,
        annotations: {},
        guide: {},
        tutorial: {},
        needsTutorial: false,
        subject: {},
        subjectsSeenThisSession: []
    }
}

it('maps state to props', () => {
    expect(mapStateToProps(initialState, {}).isSuccess).toEqual(true);
})

it('renders', () => {
    // const tree = renderer.create(
    //     <Provider store={store}>
    //         <QuestionClassifier
    //             task={task}
    //             isFetching={false}
    //             setIsFetching={jest.fn}
    //             startNewClassification={jest.fn}
    //             saveClassification={jest.fn}
    //             saveAnnotation={jest.fn}
    //             project={project}
    //             workflow={workflow}
    //             workflowID={'1'}
    //             subject={subject}
    //             subjectSizes={subjectSizes}
    //             seenThisSession={seenThisSession}
    //             navBarActions={navBarActions}
    //             classifierActions={classifierActions}
    //         />
    //     </Provider>
    // )
    // console.log(tree);
    // console.log(tree.root.findByType(AnswerButton))
})


