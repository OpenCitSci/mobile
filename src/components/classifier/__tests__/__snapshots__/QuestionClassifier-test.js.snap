// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<SwipeClassifier
  classifierActions={
    Object {
      "setClassifierTestMode": [Function],
      "startNewClassification": [Function],
    }
  }
  isFetching={false}
  navBarActions={
    Object {
      "setColorForPageToDefault": [Function],
      "setTitleForPage": [Function],
    }
  }
  project={
    Object {
      "display_name": "Awesome project",
    }
  }
  saveAnnotation={[Function]}
  saveClassification={[Function]}
  seenThisSession={Array []}
  setIsFetching={[Function]}
  startNewClassification={[Function]}
  subject={
    Object {
      "display": Object {
        "src": "blah.jpg",
      },
      "id": "23432432",
    }
  }
  subjectSizes={
    Object {
      "resizedHeight": 100,
      "resizedWidth": 100,
    }
  }
  task={
    Object {
      "answers": Array [
        Object {
          "label": "Yes",
        },
        Object {
          "label": "No",
        },
      ],
      "question": "What was that?",
    }
  }
  workflow={
    Object {
      "configuration": Object {
        "pan_and_zoom": true,
      },
      "first_task": "T0",
      "tasks": Object {
        "T0": Object {
          "answers": Array [
            Object {
              "label": "Yes",
            },
            Object {
              "label": "No",
            },
          ],
          "question": "What was that?",
        },
      },
    }
  }
  workflowID="1"
/>
`;
